### PR TITLE
[New ] Add algebra.js w/ git auto-update

### LINKED ISSUE #7971: [Request] algebra.js
**Library name:** algebra.js
**Git repository url:** https://github.com/nicolewhite/algebra.js
**npm package url(optional):** https://www.npmjs.com/package/algebra.js
**License(s):** MIT
**Official homepage:** http://algebra.js.org
**Wanna say something? Leave message here:** They dont appear to currently have a webpack/browserified version in github, but they do offer one [here](http://algebra.js.org/javascripts/algebra.min.js) as well as [make support to build one](https://github.com/nicolewhite/algebra.js#in-the-browser-1).
# 

Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by pvnr0082t**
PR for #7971 
@dannyxx001 Please help me review this pull request, thank you.

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/nicolewhite/algebra.js
- Official website (optional, not the repository): http://algebra.js.org
- NPM package url (optional): https://www.npmjs.com/package/algebra.js
- GitHub / Bitbucket popularity (required):
  - Count of stars: 995
  - Count of watchers: 42
  - Count of forks: 42
- NPM download stats (optional):
  - Downloads in the last day: 0
  - Downloads in the last week: 33
  - Downloads in the last month: 170
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [ ] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [x] Project has public repository on famous online hosting platform (or been hosted on npm
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

close #7971, cc @nicolewhite

### CODE REVIEW

**Comment by dannyxx001** [comment]
I think the PR is OK. @PeterDaveHello please help us review again, thanks.

File: ajax/libs/algebra.js/0.2.5/algebra.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+var Fraction = require('./src/fractions');
+var Expression = require('./src/expressions').Expression;
+var Equation = require('./src/equations');
+var Parser = require('./src/parser');
```
Comment (PeterDaveHello) [inline ajax/libs/algebra.js/0.2.5/algebra.js]: @pvnr0082t  @dannyxx001  Does this work in browsers? Especially you didn't commit a `src` folder.

Comment (nicolewhite) [inline ajax/libs/algebra.js/0.2.5/algebra.js]: This used to work; I believe the minify tool I was using took care of it somehow. This seems to have changed over the last few months (see https://github.com/nicolewhite/algebra.js/issues/56), but I don't know where or how. Any help is appreciated.

Comment (PeterDaveHello) [inline ajax/libs/algebra.js/0.2.5/algebra.js]: Is it possible to rebuild it or change to use other tools?

**Comment by nicolewhite** [comment]
I updated browserify and everything seems to be fine now. You can use the standalone `algebra.min.js` file in the browser. Evidence: It is loaded and used here: http://algebra.js.org/

**Comment by PeterDaveHello** [comment]
Thanks @nicolewhite

**Comment by pvnr0082t** [comment]
@dannyxx001 I grab `algebra.min.js` from http://algebra.js.org/. Please help me review again, thank you.

**Comment by PeterDaveHello** [comment]
@pvnr0082t how many versions are there? Since some of them may not work properly, maybe you should manually add other versions, and exclude the unusable versions in `.gitignore`.

**Comment by nicolewhite** [comment]
I added the version to the filename over at http://algebra.js.org/, latest is `algebra-0.2.5.min.js`. I've confirmed the file works as a standalone in both Chrome and Firefox.

**Comment by PeterDaveHello** [comment]
Any updates here?

**Comment by PeterDaveHello** [comment]
cc #9993

**Comment by nicolewhite** [comment]
I don't think there are any action items left. I just needed to update browserify. Feel free to grab the minified file at http://algebra.js.org.

**Comment by pvnr0082t** [comment]
@nicolewhite Thanks for your help. :+1:

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by PeterDaveHello** [comment]
@pvnr0082t let's not waiting for upstream?

**Comment by pvnr0082t** [comment]
@PeterDaveHello Author only provides bundled file on the homepage. The files published on NPM seems not bundled.

**Comment by nicolewhite** [comment]
Sorry for the huge delay everyone. I do not have a lot of time for side
projects anymore. 😭 I have built all of the minified files for the
previous versions and will commit them as soon as I get off the train!

**Comment by PeterDaveHello** [comment]
@nicolewhite no worries, thanks for the update!

**Comment by nicolewhite** [comment]
Hello @PeterDaveHello,

As I mentioned to @pvnr0082t in https://github.com/nicolewhite/algebra.js/issues/69, I put the files in my build/ directory here https://github.com/nicolewhite/algebra.js/tree/master/build, but I am not sure how to publish them to npm retroactively. Any help here?

**Comment by PeterDaveHello** [comment]
@nicolewhite If you have problem with npm, we also support to fetch releases from git repo, just simpily tagged a commit as a single version, and use git auto-update config here, what do you think?

**Comment by nicolewhite** [comment]
I already use tags in my git repo. What do I need to do to get that working for you guys?

**Comment by pvnr0082t** [comment]
@nicolewhite Thanks for the update. We will set auto-update config to fetch files from `build` directory. If the new version releases in the future, please just put the latest files naming without version number, `algebra.js` and `algebra.min.js`,  into the `build` directory. Thank you.

**Comment by nicolewhite** [comment]
Okay, will do. Thanks a lot!

**Comment by nicolewhite** [comment]
Hey @pvnr0082t @PeterDaveHello, what's the status on this?

**Comment by PeterDaveHello** [comment]
Ping @pvnr0082t

**Comment by sufuf3** [comment]
Ping @pvnr0082t

**Comment by pvnr0082t** [comment]
@maruilian11 @cdnjs/intern3 
1. I've added v0.2.6 w/ git auto-update config and old versions from https://github.com/nicolewhite/algebra.js/tree/master/build. 
2. Manually remove version number in filename.

Please help me review this PR, thank you.

**Comment by sufuf3** [comment]
How about mention what old version numbers you manually added in detail commit message?
What do you think?
Thank you.

**Comment by pvnr0082t** [comment]
@extend1994 I think it's fine to set auto-update config if the new version released with correct git tag and files.

**Comment by pvnr0082t** [comment]
@sufuf3 @extend1994 Thanks for your suggestion. I've refined the commit message and mentioned the versions I added in detail. Please help me review it again, thank you.

**Comment by maruilian11** [comment]
Miss v0.2.5 in commit msg, thanks.

**Comment by sufuf3** [comment]
@pvnr0082t Please add `0.2.5` in commit message.
Then i think this PR is good to me.
Thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello except mention v0.2.5 in commit message part.
I think this PR is good enough.
I'd like to help this PR, but I meet the following status.
```
Counting objects: 32513, done.
Delta compression using up to 16 threads.
Compressing objects: 100% (18473/18473), done.
Writing objects: 100% (32513/32513), 258.90 MiB | 2.61 MiB/s, done.
Total 32513 (delta 15597), reused 26672 (delta 12314)
remote: Resolving deltas: 100% (15597/15597), completed with 477 local objects.
remote: Checking connectivity: 32513, done.
To github.com:pvnr0082t/cdnjs.git
 ! [remote rejected]         8152 -> algebra.js (permission denied)
error: failed to push some refs to 'git@github.com:pvnr0082t/cdnjs.git'
```

And there is no response since 3 Oct.
How can we do for this PR?
Thanks.

**Comment by PeterDaveHello** [comment]
@sufuf3 open a new PR and close this one once you opened it.

**Comment by sufuf3** [comment]
close this PR and move #12379
cc @PeterDaveHello

**Pull request closed without merging**